### PR TITLE
Automate game state

### LINKED ISSUE #8: Automate Dramatis Personæ table
The contents of the Dramatis Personæ table can be automated from the savestring. Assign each color a player name and collect the information on who won to make them chancellor, and list the other players with proper portraits and exile status.

Perhaps we can also add citizen portraits from the citizen character art.

### PR DESCRIPTION
**Pull request by alexrutar**
This WIP is to automate the creation of all the game state files.

### CODE REVIEW

File: assets/js/oath-parser.js [hunk 1]
```diff
@@ -4,8 +4,9 @@ import * as other from './other.js'
 
 let worldDeckSize, dispossessedDeckSize, relicDeckSize;
 const format = []
-const worldSuitArray = [0,0,0,0,0,0]
-const dispSuitArray = [0,0,0,0,0,0]
+const suits = ['discord', 'hearth', 'nomad', 'arcane', 'order', 'beast']
+const worldSuitDict = {'discord': 0, 'hearth': 0, 'nomad': 0, 'arcane': 0, 'order': 0, 'beast': 0}
```
Comment (alexrutar) [inline assets/js/oath-parser.js]: @AndBM You should take a look at the changes here for the future in case you are updating the JS again later

**Comment by alexrutar** [comment]
TODO:

- [ ] Convert sites to a table, and render them in a more robust way. Maybe worth adding the site images, as well?
- [ ] Render the `#player-profiles` table automatically. This will require deciding on a data format, adding the data to `assets/js/game_data.js`, and then generating the table with this information along with the info pulled from the savestring.
- [ ] Restructure site data output from `oath_parser.js` into the form `{cradle: [site, ...], ...}` rather than an index-sensitive array.
- [x] Change suit order to alphabetical, or match board order, etc. (just modify `suits` array).

Basically, the required data should be something like a dict with keys that are the house colors, e.g.
```js
"yellow": {
  "player": "Andreas",
  "name": "House of Dawn"
}
```

**Comment by alexrutar** [comment]
Also, it would be nice if the table of player images were responsive: i.e. display the images in a single column if the page is sufficiently narrow, and in 2 columns otherwise.

**Comment by alexrutar** [comment]
The suit order can now be changed. Is there a "canonical order"? Perhaps alphabetical, or perhaps the order presented on the board itself.

**Comment by AndBM** [comment]
>The suit order can now be changed. Is there a "canonical order"? Perhaps alphabetical, or perhaps the order presented on the board itself.

There is a canonical order in the LUA files, but it doesn't really mean anything. I would actually prefer to change the order to the order on the board: Discord, Arcane, Order, Hearth, Beast, Nomad.

**Comment by alexrutar** [comment]
Also, what does the `save.name` variable mean? In the parser output.

**Comment by AndBM** [comment]
The `save.name` string is the chronicle name, in this case `"The Helichrysum Chronicle"`

**Comment by alexrutar** [comment]
Cool, I can render this into the state as well automatically.

**Comment by AndBM** [comment]
It already is, in default.html, the element with id "name".

**Comment by alexrutar** [comment]
That element no longer seems to exist? I must have deleted it somewhere, and I cannot remember when. It's missing already in the `main` branch.

Anyway I added it back now. Also, I realized it's possible to access site variables directly from the JS at render-time! This means all the configuration files don't need to be in `game_state.js` and can instead be put directly into `_config.yml`. This is really convenient from a packaging / organizational perspective.

Edit: I've realized this is a bad idea - see #11 for the proper way to do this.

**Comment by alexrutar** [comment]
Alright all done now! The code is fully running - just the calls to render the player table is hard-coded. I'll leave it to you to edit this to get the correct calls out of the savestring.

File: assets/js/script.js [hunk 2]
```diff
@@ -30,6 +30,44 @@ const oathFancyName = {
...
+    playerFig.appendChild(playerImg)
+    playerFig.appendChild(playerCaption)
+    return playerFig
+}
+// TODO: iterate over player list here, passing the color, citizenStatus, and 
+// should be something like
+// prevGameList.forEach(
+//   player => document.getElementById("player-profiles").appendChild(createPlayerProfile(player.color, player.citizenStatus))
+// )
+// and don't forget to delete the contents of the "player-profiles" div
```
Comment (alexrutar) [inline assets/js/script.js]: nevermind I already did this
